### PR TITLE
HHH-14921 + HHH-14922 Definition of the default catalog/schema on session factory creation

### PR DESCRIPTION
**Pull request by yrodiere**
* [HHH-14922](https://hibernate.atlassian.net/browse/HHH-14922): Inconsistent precedence of orm.xml implicit catalog/schema over "default_catalog"/"default_schema"
* [HHH-14921](https://hibernate.atlassian.net/browse/HHH-14921): Definition of the default catalog/schema on session factory creation

This should allow us to change the catalog/schema in "runtime" settings in Quarkus: https://github.com/quarkusio/quarkus/issues/19660

I am currently working on porting this to the 6.0 branch. It's going to take some time, considering the size of this patch.

### CODE REVIEW

**Comment by sebersole** [comment]
130 changed files and every build fails.  Mine had 5 files changed and all builds passed.  Why did you supersede mine?

**Comment by sebersole** [comment]
I created a version of my PR targetting main - https://github.com/hibernate/hibernate-orm/pull/4354

IMO that should superseded this PR at least as far as HHH-14922 is concerned.  I think we should include your work for HHH-14921 and tests on top of https://github.com/hibernate/hibernate-orm/pull/4354

**Review by sebersole — CHANGES_REQUESTED** [review]
As stated, IMO, https://github.com/hibernate/hibernate-orm/pull/4354 should supersede this PR wrt HHH-14922

**Comment by yrodiere** [comment]
> 130 changed files

Yes, that's unfortunate, but this is to address [HHH-14921](https://hibernate.atlassian.net/browse/HHH-14921): Definition of the default catalog/schema on session factory creation, which is unfortunately a much complex problem.

> every build fails

Having a look right now. It passed locally... :/

>  Why did you supersede mine?

First, because your PR didn't pass the integration tests I'm adding in this PR (very first commit). I think I explained that on Zulip, shortly.

Second, because I need a fix in 5.x, not 6.0.

But more importantly, because my actual goal is to address [HHH-14921](https://hibernate.atlassian.net/browse/HHH-14921), and THAT (much more complex) patch will require a completely different solution for [HHH-14922](https://hibernate.atlassian.net/browse/HHH-14922), unfortunately.

Let's discuss this on Zulip.

**Comment by yrodiere** [comment]
Alright, we [decided to go with this PR](https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev/topic/Default.20and.20implicit.20schema/near/261502988) for now and [apply Steve's improvements](https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev/topic/Default.20and.20implicit.20schema/near/261606256) after this PR has been merged, if he has time.

@sebersole @beikov @dreab8 could someone please approve this PR?

I have a port ready for 6.0: https://github.com/hibernate/hibernate-orm/pull/4371. So in order to merge main into 6.0, we'll just need to create a local branch where we revert all commits from this PR, then merge; then we can apply my port.

Thanks.

**Comment by yrodiere** [comment]
For the record, I got the approval for merging on Zulip: https://hibernate.zulipchat.com/#narrow/stream/132094-hibernate-orm-dev/topic/Default.20and.20implicit.20schema/near/263869953

Will merge when CI jobs go green.

**Pull request merged**
